### PR TITLE
Feature/authenticate

### PR DESCRIPTION
**Pull request by epinczinger**
The idea was to build an exclusive clubhouse where our members can write posts about non-members. Inside the clubhouse, members can see who the author of a post is but, outside, they can only see the story and wonder who wrote it.
You can check for the [project requirements here](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication). 

In the project, we practised the following concepts of rails

- Authentication
- Models
- Migrations
- Controllers
- Views
- Associations

### CODE REVIEW

**Review by mikenath223 — CHANGES_REQUESTED** [review]
### Hi Team 👋

Good work so far on this.

However, let's fix these:

- [x] Please make post_params a private method.

- [x] Consider validating the title and body field in the post table to avoid empty posts.

 - [x] Use a before_action in the posts controller to allow only authenticated users to create posts.

- [x] Please take note of this; as a guest, you can see only post's text ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/guest_can_see_only_posts_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication)). Please implement it.

Kindly fix these issues and request another review👍
Feel free to contact me on slack(michgolden) if you need further clarification.

HAPPY CODING👨‍💻

**Comment by umairarshadbutt** [comment]
**•┈┈•┈┈•⊰✿✿⊱•┈┈•┈┈•**
All changes have been done.
Thanks for the review! It was very helpful
**•┈┈•┈┈•⊰✿✿⊱•┈┈•┈┈•**

**Comment by lekegitrepo** [comment]
> **•┈┈•┈┈•⊰✿✿⊱•┈┈•┈┈•**
> All changes have been done.
> Thanks for the review! It was very helpful
> **•┈┈•┈┈•⊰✿✿⊱•┈┈•┈┈•**

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+class PostsController < ApplicationController
```
Comment (lekegitrepo) [inline app/controllers/posts_controller.rb:1]: * [x] As mentioned in the previous review: "Use a before_action in the posts controller to allow only authenticated users to create posts."

**Review by lekegitrepo — CHANGES_REQUESTED** [review]
### Status: Request changes :recycle:
#### You have done a good job so far :+1: however, you forgot to implement one of the required changes mentioned in the previous review. Check my comment below :point_down:

#### _Reviewed by Temitayo Adeleke._

**Comment by umairarshadbutt** [comment]
**•┈┈•┈┈•┈┈•┈┈•┈┈•⊰✿✿⊱•┈┈•┈┈•┈┈•┈┈•┈┈•**
🔘 Use of before_action in the posts controller
**•┈┈•┈┈•┈┈•┈┈•┈┈•⊰✿✿⊱•┈┈•┈┈•┈┈•┈┈•┈┈•**

**Review by rOluochKe — APPROVED** [review]
Hello Team
### A well done job, you got it right :clap: The project has been approved. Congratulations :boom: :100:
### Merge and submit the project. Good luck in the next project.

**Comment by epinczinger** [comment]
Thanks for the review!

**Pull request merged**